### DISCUSSION TITLE
Will Next.js drop Webpack support in the future in favor of Turbopack?

### DISCUSSION THREAD

**Discussion by drixsonic** [vercel/next.js#51278] [Help]
### Summary

I'm currently building some prototypes using the [nextjs-mf plugin](https://github.com/module-federation/universe/tree/main/packages/nextjs-mf), and I'm wondering how future-proof they are. We're exploring the possibility to use composable decoupled frontends and if we want to use Next.js, the use of that plugin seems mandatory if we want the apps to be deployed independently. 

But if we go in that direction and Next.js drops Webpack support, and with it the possibility of using Module Federation, our solution is blocked on certain Next.js version. 

So my question here is: what are the plans concerning Webpack at Vercel? If Turbopack is meant to replace Webpack, would we still be given the choice to choose the one we prefer/need?

**Comment by kwafoawua** [comment]
Hi there,
With Next.js 16 making Turbopack the default bundler for both development and production builds, I’m wondering about the future of Webpack support. 

Specifically:
1. Is there a plan to deprecate or remove Webpack support entirely in future Next.js versions?
2. If so, what version or timeline should we expect this to happen?
3. Will there still be an official way to build with Webpack (next dev --webpack, next build --webpack) after Next.js 16?

Thanks!
